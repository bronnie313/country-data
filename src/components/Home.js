import React from 'react';
import CountryList from './countryList';

const Home = () => (
  <CountryList />
);

export default Home;
