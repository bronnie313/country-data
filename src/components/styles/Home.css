.image {
    width: 20%;
}

.home-img {
    background: #fd5294;
    display: flex;
    justify-content: center;
    align-items: center;
    color: white;
}

.stats {
    display: flex;
    color: #fff;
    background: #e04783;
    padding: 1rem;
    font-size: smaller;
}