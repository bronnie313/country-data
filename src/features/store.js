import { configureStore } from '@reduxjs/toolkit';
import modalReducer from './modal/modalSlice';
import countryReducer from './countries/countriesSlice';

const store = configureStore({
  reducer: {
    modal: modalReducer,
    country: countryReducer,
  },
});

export default store;
