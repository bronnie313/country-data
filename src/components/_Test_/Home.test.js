import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../../features/store';
import Home from '../Home';

describe('home component', () => {
  it('testing if page matches', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Home />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
