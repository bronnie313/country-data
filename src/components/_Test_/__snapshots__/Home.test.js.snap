// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`home component testing if page matches 1`] = `
<div>
  <div
    className="home-img"
  >
    <img
      alt="Flag"
      className="image"
      src="world.png"
    />
    <div
      className="world"
    >
      <h1>
        WORLD STATISTICS
      </h1>
    </div>
  </div>
  <div
    className="stats"
  >
    <h2>
      STATS BY COUNTRY
    </h2>
    <div>
      <input
        className="search-input"
        id="search"
        name="search"
        onChange={[Function]}
        placeholder="search counrty"
        type="search"
      />
    </div>
  </div>
  <section>
    <div
      className="countries"
    />
  </section>
</div>
`;
